Wikipedia/Code/MWKDataStore.m

```diff
@@ -518,13 +537,35 @@ - (void)performLibraryUpdates:(dispatch_block_t)completion {
         dispatch_block_t done = ^{
             dispatch_async(dispatch_get_main_queue(), completion);
         };
-        if (currentLibraryVersion < libraryVersion) {
-            [self performUpdatesFromLibraryVersion:currentLibraryVersion inManagedObjectContext:moc];
-        }
+        [self performUpdatesFromLibraryVersion:currentUserLibraryVersion inManagedObjectContext:moc];
         done();
     }];
 }
 
+- (void)markAllDownloadedArticlesInManagedObjectContextAsNeedingConversionFromMobileview:(NSManagedObjectContext *)moc {
+    NSFetchRequest *request = [WMFArticle fetchRequest];
+    request.predicate = [NSPredicate predicateWithFormat:@"isDownloaded == YES"];
+    request.propertiesToFetch = @[];
+    NSError *fetchError = nil;
+    NSArray *downloadedArticles = [moc executeFetchRequest:request error:&fetchError];
```

Comment:
This fetch should be batched similar to the migrations below to prevent an excessively large fetch result for users with thousands of saved articles